vaadin-icons-flow-parent/vaadin-icons-flow/src/test/java/com/vaadin/flow/component/icon/tests/IconTest.java

```diff
@@ -19,12 +19,50 @@
 import org.junit.Test;
 
 import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
 import com.vaadin.flow.component.shared.HasTooltip;
 
 public class IconTest {
+
     @Test
     public void implementsHasTooltip() {
         Icon icon = new Icon();
         Assert.assertTrue(icon instanceof HasTooltip);
     }
+
+    @Test
+    public void emptyIconIsEmpty() {
+        Assert.assertNull(new Icon().getIcon());
+    }
+
+    @Test
+    public void usesVaadinCollectionByDefault() {
+        Assert.assertEquals("vaadin:foo", new Icon("foo").getIcon());
+    }
+
+    @Test
+    public void canDefineCollectionInConstructor() {
+        Assert.assertEquals("bar:foo", new Icon("bar:foo").getIcon());
+    }
+
+    @Test
+    public void canDefineCollectionInSetter() {
+        Icon icon = new Icon();
+        icon.setIcon("bar:foo");
+        Assert.assertEquals("bar:foo", icon.getIcon());
+    }
+
+    @Test
+    public void setterUsesCurrentCollection() {
+        Icon icon = new Icon("bar:foo");
+        icon.setIcon("baz");
+        Assert.assertEquals("bar:baz", icon.getIcon());
+    }
+
+    @Test
+    public void canSetNewVaadinIcon() {
```

Comment:
**nit:** It would be nice to have a test for `getCollection` since it's also a public method.